src/Storages/ExternalStream/StorageExternalStreamImpl.h

```diff
@@ -12,7 +14,11 @@ namespace DB
 class StorageExternalStreamImpl : public std::enable_shared_from_this<StorageExternalStreamImpl>
 {
 public:
-    explicit StorageExternalStreamImpl(std::unique_ptr<ExternalStreamSettings> settings_): settings(std::move(settings_)) {
+    explicit StorageExternalStreamImpl(IStorage * storage, std::unique_ptr<ExternalStreamSettings> settings_, const ContextPtr & context)
+        : storage_id(storage->getStorageID())
+        , settings(std::move(settings_))
+        , tmpdir(fs::path(context->getConfigRef().getString("tmp_path", fs::temp_directory_path())) / "external_streams" / toString(storage_id.uuid))
```

Comment:
If we fallback to fs::temp_..., is it safe? Since after reboot, everything is gone ?

Reply:
Yes, this is safe, because the cert pem file will be created again after reboot.